elyra/pipeline/tests/test_pipeline_parser.py

```diff
@@ -127,18 +108,68 @@ def test_pipeline_operations_and_handle_artifact_file_details():
 
 
 def test_pipeline_with_dependencies():
-    pipeline_definition = read_pipeline_resource('pipeline_3_node_sample_with_dependencies.json')
+    pipeline_definition = __read_pipeline_resource('pipeline_3_node_sample_with_dependencies.json')
 
     pipeline = PipelineParser.parse(pipeline_definition)
 
     assert len(pipeline.operations['acc4527d-7cc8-4c16-b520-5aa0f50a2e34'].dependencies) == 2
 
 
 def test_pipeline_global_attributes():
-    pipeline_definition = read_pipeline_resource('pipeline_valid.json')
+    pipeline_definition = __read_pipeline_resource('pipeline_valid.json')
 
     pipeline = PipelineParser.parse(pipeline_definition)
 
     assert pipeline.title == '{{title}}'
     assert pipeline.runtime == '{{runtime}}'
     assert pipeline.runtime_config == '{{runtime-config}}'
+
+
+def test_missing_operation_id():
+    pipeline_definition = __read_pipeline_resource('pipeline_valid.json')
+    pipeline_definition['pipelines'][0]['nodes'][0]['id'] = ''
+
+    with pytest.raises(SyntaxError) as e:
+        PipelineParser.parse(pipeline_definition)
+
+    assert "Missing field 'operation id'" in str(e.value)
+
+
+def test_missing_operation_type():
+    pipeline_definition = __read_pipeline_resource('pipeline_valid.json')
+    pipeline_definition['pipelines'][0]['nodes'][0]['type'] = ''
+
+    with pytest.raises(SyntaxError) as e:
+        PipelineParser.parse(pipeline_definition)
+
+    assert "Missing field 'operation type'" in str(e.value)
+
+
+def test_missing_operation_artifact():
+    pipeline_definition = __read_pipeline_resource('pipeline_valid.json')
+    pipeline_definition['pipelines'][0]['nodes'][0]['app_data']['artifact'] = ''
+
+    with pytest.raises(SyntaxError) as e:
+        PipelineParser.parse(pipeline_definition)
+
+    assert "Missing field 'operation artifact" in str(e.value)
+
+
+def test_missing_operation_image():
+    pipeline_definition = __read_pipeline_resource('pipeline_valid.json')
+    pipeline_definition['pipelines'][0]['nodes'][0]['app_data']['image'] = ''
+
+    with pytest.raises(SyntaxError) as e:
+        PipelineParser.parse(pipeline_definition)
+
+    assert "Missing field 'operation image" in str(e.value)
+
+
+def __read_pipeline_resource(pipeline_filename):
```

Comment:
Shouldn't this be a single underscore?  At least I thought that was the convention for "local/private" functions in python.